test/sql-luatest/collation_test.lua

```diff
@@ -1,31 +1,865 @@
 local server = require('luatest.server')
 local t = require('luatest')
 
-local g = t.group()
+local g = t.group("collation", {{engine = 'memtx'}, {engine = 'vinyl'}})
 
-g.before_all(function()
-    g.server = server:new({alias = 'master'})
-    g.server:start()
+g.before_each(function(cg)
+    cg.server = server:new({alias = 'master'})
+    cg.server:start()
+    cg.server:exec(function(engine)
+        box.execute([[SET SESSION "sql_seq_scan" = true;]])
+        local sql = [[SET SESSION "sql_default_engine" = '%s']]
+        box.execute(sql:format(engine))
+    end, {cg.params.engine})
 end)
 
-g.after_all(function()
-    g.server:stop()
+g.after_each(function(cg)
+    cg.server:drop()
 end)
 
-g.test_ghs_80 = function()
-    g.server:exec(function()
+g.test_collation = function(cg)
+    cg.server:exec(function()
+        -- Explicitly set BINARY collation is predefined and has ID.
+        box.execute([[CREATE TABLE t (id INT PRIMARY KEY,
+                      a TEXT, b TEXT COLLATE "binary");]])
+        t.assert(box.space.t ~= nil)
+        local res = box.space.t:format()[2]['collation']
+        t.assert_equals(res, nil)
+        res = box.space.t:format()[3]['collation']
+        t.assert_equals(res, 3)
```

Comment:
It is not actually obvious what this `3` is. Let's use `binary` collation ID instead here and below.

For example, you can do this:
```Lua
        local coll_binary_id = box.space._collation.index.name:get('binary').id
        t.assert_equals(res, coll_binary_id)
```
or this:
```Lua
        local coll_binary_id = box.internal.collation.id_by_name('binary')
        t.assert_equals(res, coll_binary_id)
```
or use some other method to get collation ID.